shuffle/cli/src/shared.rs

```diff
@@ -173,14 +173,28 @@ impl DevApiClient {
     }
 
     fn check_accounts_resources_response_status_code(status_code: &StatusCode) -> Result<()> {
-        match status_code == &StatusCode::from_u16(200)? {
+        match status_code == &StatusCode::OK {
             true => Ok(()),
             false => Err(anyhow!(
                 "Failed to get account resources with provided address"
             )),
         }
     }
 
+    pub fn check_post_transactions_response_status(status_code: &StatusCode) -> Result<()> {
+        match status_code == &StatusCode::ACCEPTED {
+            true => Ok(()),
+            false => Err(anyhow!("Failed to post transactions")),
+        }
+    }
+
+    pub fn check_get_transactions_response_status(status_code: &StatusCode) -> Result<()> {
```

Comment:
Let's collapse `check_get_transactions_response_status` and `check_post_transactions_response_status` into simply `check_response_status`. They both have available a `Response` instance, and that can used to have an insightful error message, since it has headers, probably the verb, and other goodies.

```rust
    let mut resp = client.get_transactions_by_hash(hash).await?;
    DevApiClient::check_response_status(&resp)?;
```

Reply:
Hmm. I agree that I can pass in resp into `check_response_status` to print out more info in error messages, but I think we should separate out check_get_transactions_response_status and check_post_transactions_response_status since they have different statuscodes. For POST /transactions it's ACCEPTED (202) and for GET /transactions it's OK (200)